build fails on s390x due too many arguments for mono_handle_native_crash()

The recent revert in commit 2a4e66af4dc628a84a42ff443bc2c09a4bcc7dec breaks build on s390x, because it doesn't revert the change from commit 1d3e43ec3a5d2b9f672d86b12fce84a8da654cb6
````
make[4]: Entering directory '/home/jenkins/workspace/Mono/label/s390x/mono/mini'
  CC       libmini_la-exceptions-s390x.lo
exceptions-s390x.c: In function ‘mono_arch_get_call_filter’:
exceptions-s390x.c:123:15: warning: request for implicit conversion from ‘void *’ to ‘guint8 *’ {aka ‘unsigned char *’} not permitted in C++ [-Wc++-compat]
  123 |  code = start = mono_global_codeman_reserve (512);
      |               ^
exceptions-s390x.c: In function ‘mono_arch_get_throw_exception_generic’:
exceptions-s390x.c:311:15: warning: request for implicit conversion from ‘void *’ to ‘guint8 *’ {aka ‘unsigned char *’} not permitted in C++ [-Wc++-compat]
  311 |  code = start = mono_global_codeman_reserve(size);
      |               ^
exceptions-s390x.c: In function ‘mono_arch_unwind_frame’:
exceptions-s390x.c:526:73: warning: request for implicit conversion from ‘gpointer’ {aka ‘void *’} to ‘guint8 *’ {aka ‘unsigned char *’} not permitted in C++ [-Wc++-compat]
  526 |   gboolean success = mono_unwind_frame (unwind_info, unwind_info_len, ji->code_start,
      |                                                                       ~~^~~~~~~~~~~~
exceptions-s390x.c:528:10: warning: request for implicit conversion from ‘gpointer’ {aka ‘void *’} to ‘guint8 *’ {aka ‘unsigned char *’} not permitted in C++ [-Wc++-compat]
  528 |          ip, epilog ? &epilog : NULL, regs, 16, save_locations,
      |          ^~
exceptions-s390x.c:507:13: warning: variable ‘address’ set but not used [-Wunused-but-set-variable]
  507 |   uintptr_t address;
      |             ^~~~~~~
exceptions-s390x.c:556:8: warning: request for implicit conversion from ‘gpointer’ {aka ‘void *’} to ‘MonoLMF *’ {aka ‘struct MonoLMF *’} not permitted in C++ [-Wc++-compat]
  556 |   *lmf = (*lmf)->previous_lmf;
      |        ^
exceptions-s390x.c: In function ‘altstack_handle_and_restore’:
exceptions-s390x.c:576:4: error: too many arguments to function ‘mono_handle_native_crash’
  576 |    mono_handle_native_crash (mono_get_signame (SIGSEGV), ctx, siginfo, ctx);
      |    ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from exceptions-s390x.c:63:
mini.h:2523:10: note: declared here
 2523 | void     mono_handle_native_crash               (const char *signal, MonoContext *mctx, MONO_SIG_HANDLER_INFO_TYPE *siginfo);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
make[4]: *** [Makefile:2497: libmini_la-exceptions-s390x.lo] Error 1
````

CC @nealef

--- a/src/mono/mono/mini/exceptions-s390x.c
+++ b/src/mono/mono/mini/exceptions-s390x.c
@@ -573,7 +573,7 @@ altstack_handle_and_restore (MonoContext *ctx, MONO_SIG_HANDLER_INFO_TYPE *sigin
 
 	if (!ji || (!stack_ovf && !nullref)) {
 		if (mono_dump_start ())
-			mono_handle_native_crash (mono_get_signame (SIGSEGV), ctx, siginfo, ctx);
+			mono_handle_native_crash (mono_get_signame (SIGSEGV), ctx, siginfo);
 		/* if couldn't dump or if mono_handle_native_crash returns, abort */
 		abort ();
 	}

